base/shared/src/main/scala/scalaz/data/Forall.scala

```diff
@@ -48,26 +48,13 @@ trait ForallModule {
   object Unapply {
     implicit def unapply[G[_]]: Unapply[∀[G]] { type F[A] = G[A] } =
       new Unapply[∀[G]] { type F[A] = G[A] }
-
-    implicit def unapply1[G[_], H[_]]: Unapply[∀[λ[α => G[H[α]]]]] { type F[A] = G[H[A]] } =
-      new Unapply[∀[λ[α => G[H[α]]]]] { type F[A] = G[H[A]] }
-
-    implicit def unapply2[P[_, _]]: Unapply[∀[λ[α => P[α, α]]]] { type F[A] = P[A, A] } =
-      new Unapply[∀[λ[α => P[α, α]]]] { type F[A] = P[A, A] }
-
-    implicit def unapply3[P[_, _], G[_], H[_]]: Unapply[∀[λ[α => P[G[α], H[α]]]]] { type F[A] = P[G[A], H[A]] } =
-      new Unapply[∀[λ[α => P[G[α], H[α]]]]] { type F[A] = P[G[A], H[A]] }
   }
 }
 
 trait ForallSyntax {
   import ForallSyntax._
 
   implicit def toForallOps[F[_]](a: ∀[F]): Ops[F] = new Ops[F](a)
```

Comment:
Ditto.

Reply:
Likewise, plus scalac now contains some very similar [test cases](https://github.com/scala/scala/blob/c2a5883891a68180b143eb462c8b0cebc8d3b021/test/files/run/hk-typevar-unification.scala) for these. 😈

Reply:
❤️ ❤️ ❤️